method.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return true unless block_given?
+
+    (0...length).each do |i|
+      yield(self[i])
+    end
+  end
+
+  def my_each_with_index
+    return true unless block_given?
+
+    k = 0
+    (0...length).each do |i|
+      yield(self[i], k)
+      k += 1
+    end
+  end
+
+  def my_select(&block)
+    return true unless block_given?
+
+    result = []
+    each do |element|
+      result << element if block.call(element) == true
+    end
+    result
+  end
+
+  def my_all?(pattern = nil)
+    if block_given?
+      my_each { |v| return false unless yield(v) }
+    elsif !pattern.nil?
+      my_each { |v| return false unless pattern == v.to_i }
+    else
+      my_each { |v| return false unless v }
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
+    if block_given?
+      my_each { |v| return true if yield(v) }
+    elsif !pattern.nil?
+      my_each { |v| return true if pattern == v.to_i }
+    else
+      my_each { |v| return true if v }
+      return false
+    end
+    false
+  end
+
+  def my_none?(pattern = nil)
+    if block_given?
+      my_each { |v| return false if yield(v) }
+    elsif !pattern.nil?
+      my_each { |v| return false if pattern == v.to_i }
+    else
+      my_each { |v| return false if v }
+    end
+    true
+  end
```

Comment:
1. When a class is passed as an argument returns true if none of the collection is a member of such class

`[1, 3.14, 42].my_none?(Integer)` should return false
`[1, 3.14, 42].my_none?(String)` should return true

2. When a Regex is passed as an argument returns true only if none of the collection matches the Regex

`['ant', 'bear', 'boy'].my_none?(/t/)` shoud returns false
`['car', 'bear', 'boy'].my_none?(/t/)` shoud returns true